### PR TITLE
RATIS-1611. Add notify server step sown event to inform

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/browse/RATIS-1611

### CODE REVIEW

**Comment by codings-dan** [comment]
@szetszwo Can you help review this pull request, thank you in advance.

File: ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java [hunk 1]
```diff
@@ -192,6 +192,11 @@ default void notifyLogFailed(Throwable cause, LogEntryProto failedEntry) {}
      * completely done. Could trigger the cleanup of snapshots.
      */
     default void notifySnapshotInstalled(InstallSnapshotResult result, long snapshotIndex) {}
+
+    /**
+     * Notify the {@link StateMachine} that a raft server has step down.
+     */
+    default void notifyServerStepDown() {}
```
Comment (szetszwo) [inline ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java:199]: Let's call it notifyServerShutdown and pass a RoleInfoProto parameter.

Comment (codings-dan) [reply ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java]: done

**Comment by codings-dan** [comment]
@szetszwo PTAL again, thank you!

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**